File: src/2395. Find Subarrays With Equal Sum.cpp
class Solution {
public:
    bool findSubarrays(vector<int>& nums) {
        int n = nums.size();
        set<int> existed;
        for (int i = 1; i < n; ++i) {
            int sum = nums[i] + nums[i - 1];
            if (existed.count(sum)) return true;
            existed.insert(sum);
        }
        return false;
    }
};

File: src/2395. Find Subarrays With Equal Sum.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
function findSubarrays(nums) {
  const n = nums.length,
    set = new Set()
  for (let i = 1; i < n; i++) {
    const sum = nums[i] + nums[i - 1]
    if (set.has(sum)) return true
    set.add(sum)
  }
  return false
}
